src/headers/mod.rs

```diff
@@ -0,0 +1,39 @@
+use async_trait::async_trait;
+use reqwest::header::HeaderMap;
+use std::convert::Infallible;
+
+mod lnurl_auth_jwt;
+pub use lnurl_auth_jwt::LnurlAuthJwt;
+
+/// Defines a trait around how headers are provided for each VSS request.
+#[async_trait]
+pub trait HeaderProvider {
+	/// Error type when returning headers.
+	type Error: std::error::Error;
```

Comment:
What do you have in mind here? I added the associated type with a trait bound because each implementation might run into different kinds of errors. Returning fixed headers is infallible, but LNURL Auth might run into protocol-specific errors.

Reply:
Simplified these into some basic error cases, but let me know if using `String` or `std::io::Error` would be better.